src/lib/actionrunner.js

```diff
@@ -287,6 +288,11 @@ class OpenwhiskActionRunner {
     _getImage() {
         // blackbox -> image is specified directly
         if (this.action.exec.image) {
+            if(process.env.ASSET_COMPUTE_DOCKER_IMAGE_PREFIX 
+                && validUrl.isHttpsUri(process.env.ASSET_COMPUTE_DOCKER_IMAGE_PREFIX)){
+                // for blackbox containers that may grab images from repositories depending on region
+                return process.env.ASSET_COMPUTE_DOCKER_IMAGE_PREFIX + this.action.exec.image;
```

Comment:
I am confused - we already handle this [here in the nui cli](https://git.corp.adobe.com/nui/cli/blob/493b4a4ab0f1af5a977060fe904cf63afa083437/lib/openwhisk.js#L322-L330).

What we currently do:
- all workers have the docker registry in their image name in package.json
- this works for local wskdebug & test-worker (code here)
- upon deployment, it gets [removed](https://git.corp.adobe.com/nui/cli/blob/493b4a4ab0f1af5a977060fe904cf63afa083437/lib/openwhisk.js#L322-L330) as Runtime no longer accepts a registry in there anymore ([since July 2020](https://wiki.corp.adobe.com/display/AdobeCloudPlatform/Creating+Blackbox+Actions+for+IO+Runtime))

Based on that, why is this change even required?

Reply:
Doing it the other way around (no prefix in sources, only add for local testing) is tricky as it wouldn't work for the generic `wskdebug` which needs to pull the image as well but has no idea what registry different images might use.